extensions/tasks-parent/tasks-client/src/main/java/io/resys/thena/tasks/client/spi/visitors/TaskCommandVisitor.java

```diff
@@ -246,6 +229,33 @@ private Task visitChangeTaskExtension(ChangeTaskExtension command) {
     return this.current;
   }
 
+  private Task visitChangeTaskExtension(ChangeTaskExtension command) {
+    final var id = Optional.ofNullable(command.getId()).orElseThrow(() -> new UpdateTaskVisitorException("Extension id not found"));
+    final var oldExtension = current.getExtensions().stream()
```

Comment:
You're iterating over the extensions twice. 

Combine this exception with a for loop via some boolean variable (whether the task was found or not).